rts/motoko-rts/src/gc/mark_compact.rs

```diff
@@ -57,18 +57,18 @@ unsafe fn compacting_gc<M: Memory>(mem: &mut M) {
         |reclaimed| linear_memory::RECLAIMED += Bytes(u64::from(reclaimed.as_u32())),
     );
 
-    linear_memory::LAST_HP = linear_memory::getHP();
+    linear_memory::LAST_HP = linear_memory::get_hp_unskewed();
 }
 
 pub unsafe fn compacting_gc_internal<
     M: Memory,
     GetHp: Fn() -> usize,
-    SetHp: Fn(u32),
+    SetHp: Fn(usize),
     NoteLiveSize: Fn(Bytes<u32>),
     NoteReclaimed: Fn(Bytes<u32>),
 >(
     mem: &mut M,
-    heap_base: u32,
+    heap_base: usize,
     get_hp: GetHp,
```

Comment:
Should we maybe rename this helper function, because it returns the unskewed pointer and has the same name like the skewed `get_hp()` in the compiler-generated code? The same would apply for `set_hp()`, maybe also in the other classical GCs?

Reply:
`getHP` and `setHP` are confined into the source file where they are imported, I don't see danger of confusion. But see my other comments.